packages/headless/src/features/history/history-slice.ts

```diff
@@ -47,6 +47,7 @@ const isEqual = (current: HistoryState, next: HistoryState) => {
 const isContextEqual = (current: ContextState, next: ContextState) =>
   JSON.stringify(current.contextValues) === JSON.stringify(next.contextValues);
 
+// TODO: compare all facets non-idle values, comparing current values changes history after 1st request
```

Comment:
That's another problem, when doing the "same" request twice initially, the first one will have no current value, the following one will, which makes both history state not equal, when in fact they are.